src/Microsoft.Data.Analysis/PrimitiveColumnContainer.cs

```diff
@@ -127,31 +128,40 @@ public PrimitiveColumnContainer(ReadOnlyMemory<byte> buffer, ReadOnlyMemory<byte
 
         public PrimitiveColumnContainer(long length = 0)
         {
-            while (length > 0)
+            /*
+            var remaining = length;
+            while (remaining > 0)
             {
                 if (Buffers.Count == 0)
                 {
                     Buffers.Add(new DataFrameBuffer<T>());
                     NullBitMapBuffers.Add(new DataFrameBuffer<byte>());
                 }
-                DataFrameBuffer<T> lastBuffer = (DataFrameBuffer<T>)Buffers[Buffers.Count - 1];
-                if (lastBuffer.Length == ReadOnlyDataFrameBuffer<T>.MaxCapacity)
+
+                if (Buffers[Buffers.Count - 1].Length == ReadOnlyDataFrameBuffer<T>.MaxCapacity)
                 {
-                    lastBuffer = new DataFrameBuffer<T>();
-                    Buffers.Add(lastBuffer);
+                    Buffers.Add(new DataFrameBuffer<T>());
                     NullBitMapBuffers.Add(new DataFrameBuffer<byte>());
                 }
-                int allocatable = (int)Math.Min(length, ReadOnlyDataFrameBuffer<T>.MaxCapacity);
-                lastBuffer.EnsureCapacity(allocatable);
-                DataFrameBuffer<byte> lastNullBitMapBuffer = (DataFrameBuffer<byte>)(NullBitMapBuffers[NullBitMapBuffers.Count - 1]);
+
+                DataFrameBuffer<T> mutableLastBuffer = Buffers.GetMutable(Buffers.Count - 1);
+
+                int allocatable = (int)Math.Min(remaining, ReadOnlyDataFrameBuffer<T>.MaxCapacity);
+
+                mutableLastBuffer.EnsureCapacity(allocatable);
+                mutableLastBuffer.Length = allocatable;
+
+                DataFrameBuffer<byte> lastNullBitMapBuffer = NullBitMapBuffers.GetMutable(NullBitMapBuffers.Count - 1);
                 int nullBufferAllocatable = (allocatable + 7) / 8;
                 lastNullBitMapBuffer.EnsureCapacity(nullBufferAllocatable);
-                lastBuffer.Length = allocatable;
                 lastNullBitMapBuffer.Length = nullBufferAllocatable;
-                length -= allocatable;
-                Length += lastBuffer.Length;
-                NullCount += lastBuffer.Length;
-            }
+
+                remaining -= allocatable;
+                Length += mutableLastBuffer.Length;
+                NullCount += mutableLastBuffer.Length;
+            }*/
+
+            AppendMany(null, length);
```

Comment:
I think all the code commented out can be removed. It's the same as calling AppendMany with null.

Reply:
You aren't setting Length or NullCount then which will potentially cause problems since those are public variables.

Reply:
AppendMany does that though.